### PR TITLE
PlanBuilder related cleanups

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java [hunk 1]
```diff
@@ -67,14 +53,9 @@ public void testPushesAggregationThroughLeftJoin()
...
-                                ),
-                                ImmutableMap.of(p.symbol("AVG", DOUBLE),
-                                        new AggregationNode.Aggregation(averageFunctionCall, avgSignature, Optional.empty())),
-                                ImmutableList.of(ImmutableList.of(p.symbol("COL1", BIGINT))),
-                                SINGLE,
-                                Optional.empty(),
-                                Optional.empty())
-                )
+                                ))
+                        .addAggregation(p.symbol("AVG", DOUBLE), PlanBuilder.expression("avg(COL2)"), ImmutableList.of(DOUBLE))
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java]: static import `expression`?

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java [hunk 2]
```diff
@@ -67,14 +53,9 @@ public void testPushesAggregationThroughLeftJoin()
...
+                        .groupingSets(ImmutableList.of(ImmutableList.of(p.symbol("COL1", BIGINT))))))
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java]: Looks awful. Maybe we could have `.groupBy(Symbol... )` for cases like this?
That would remove need for `ImmutableList.of(ImmutableList.of(...))`.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java (hunk 1 above)
Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java]: Already different `expression` is static imported.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java (hunk 2 above)
Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java]: Yeah - a place for further extension.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java [hunk 3]
```diff
@@ -46,18 +41,9 @@
...
-                FunctionKind.AGGREGATE,
-                ImmutableList.of(),
-                ImmutableList.of(),
-                DOUBLE.getTypeSignature(),
-                ImmutableList.of(DOUBLE.getTypeSignature()),
-                false);
         new RuleTester().assertThat(new PushAggregationThroughOuterJoin())
-                .on(p ->
-                        p.aggregation(
+                .on(p -> p.aggregation(ab -> ab
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java]: can we somehow avoid using nested lambda? Maybe by a method extraction?

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java (hunk 2 above)
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java]: or maybe we could call a `groupingSet` method per each grouping set and then builder would collect them internally
`
.groupingSet(p.symbol("C1"))
.groupingSet(p.symbol("C2"), p.symbol("C1")
`

How does it sound to you?

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestTransformCorrelatedScalarAggregationToJoin.java [hunk 4]
```diff
@@ -147,14 +139,11 @@ public void rewritesOnSubqueryWithProjection()
                                                 values(ImmutableMap.of("a", 0, "b", 1)))))));
     }
 
-    private AggregationNode createSumAggregation(PlanBuilder p, Symbol symbol, List<List<Symbol>> groupingSets, PlanNode source)
+    private AggregationNode createSumAggregation(PlanBuilder p, List<List<Symbol>> groupingSets, PlanNode source)
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestTransformCorrelatedScalarAggregationToJoin.java]: I wonder if the method invocation is not more difficult to understand than then method body. I would inline this method then, thanks to that there would be no need to pass empty List<List> arguments as there will be no need to call `groupingSets` method

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java (hunk 3 above)
Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java]: You could achieve that by having 
`PlanBuilder.onAggregation(BiConsumer<PlanBuilder, AggreagtionBuilder>` but this is very usecase specific and I do not believe worth pursuing.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java (hunk 2 above)
Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java]: I added `addGroupingSet` method and used in applicable places.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestTransformCorrelatedScalarAggregationToJoin.java (hunk 4 above)
Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestTransformCorrelatedScalarAggregationToJoin.java]: +1. It looks nicer to me.

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java [hunk 5]
```diff
@@ -160,7 +160,17 @@ public AggregationBuilder globalGrouping()
 
         public AggregationBuilder groupingSets(List<List<Symbol>> groupingSets)
         {
-            this.groupingSets = ImmutableList.copyOf(groupingSets);
+            this.groupingSets.clear();
```
Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java]: Instead you could check `this.groupingSets.isEmpty()`

Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java]: And only allow calling `groupingSets` if the list is empty?

Comment (findepi) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java]: Yes, that was my intention.

Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java]: updated

**Pull request closed without merging**